extras/kinesis-asl/src/main/scala/org/apache/spark/streaming/kinesis/KinesisReceiver.scala

```diff
@@ -70,6 +69,20 @@ private[kinesis] class KinesisReceiver(
     storageLevel: StorageLevel)
   extends Receiver[Array[Byte]](storageLevel) with Logging { receiver =>
 
+  def this(appName: String,
+           streamName: String,
+           endpointUrl: String,
+           checkpointInterval: Duration,
+           initialPositionInStream: InitialPositionInStream,
+           storageLevel: StorageLevel,
+           credentials: AWSCredentials) = {
```

Comment:
Since this is internal, can this just be implemented with an optional final parameter to the existing declared constructor?